src/saltext/vmware/states/vmc_dhcp_profiles.py

```diff
@@ -0,0 +1,420 @@
+
+    HAS_DHCP_PROFILES = True
+except ImportError:
+    HAS_DHCP_PROFILES = False
+
+
+def __virtual__():
+    if not HAS_DHCP_PROFILES:
+        return False, "'vmc_dhcp_profiles' binary not found on system"
+    return "vmc_dhcp_profiles"
+
+
+def present(
+    name,
+    hostname,
+    refresh_key,
+    authorization_host,
+    org_id,
+    sddc_id,
+    type,
+    dhcp_profile_id,
+    verify_ssl=True,
+    cert=None,
+    server_addresses=None,
+    tags=vmc_constants.VMC_NONE,
+    lease_time=None,
+    display_name=None,
+):
+    """
+    Ensure a given DHCP profile exists for given SDDC
+
+    hostname
+        The host name of NSX-T manager
+
+    refresh_key
+        API Token of the user which is used to get the Access Token required for VMC operations
+
+    authorization_host
+        Hostname of the VMC cloud console
+
+    org_id
+        The Id of organization to which the SDDC belongs to
+
+    sddc_id
+        The Id of SDDC for which the DHCP profile should be added
+
+    type
+        The type of DHCP profile for which the given dhcp belongs to. Possible values: server, relay
+
+    dhcp_profile_id
+        Id of the DHCP profile to be added to SDDC
+
+    verify_ssl
+        (Optional) Option to enable/disable SSL verification. Enabled by default.
+        If set to False, the certificate validation is skipped.
+
+    cert
+        (Optional) Path to the SSL client certificate file to connect to VMC Cloud Console.
+        The certificate can be retrieved from browser.
+
+    server_addresses
+        when type is relay, this field indicates DHCP relay addresses(DHCP server IP addresses
+        for DHCP relay configuration).
+        Both IPv4 and IPv6 addresses are supported.
+
+        when type is server, this field indicates DHCP server address in CIDR format.
+        Both IPv4 and IPv6 address families are supported.
+        Prefix length should be less than or equal to 30 for IPv4 address family
+        and less than or equal to 126 for IPv6.
+        When not specified, IPv4 value is auto-assigned to 100.96.0.1/30.
+
+        Note: This field is optional only when type is server and
+        is mandatory when type is relay.
+
+    tags
+        (Optional) Opaque identifiers meaningful to the user.
+
+        .. code::
+
+            tags='[
+                {
+                    "tag": "<tag-key-1>"
+                    "scope": "<tag-value-1>"
+                },
+                {
+                    "tag": "<tag-key-2>"
+                    "scope": "<tag-value-2>"
+                }
+            ]'
+
+    lease_time
+        (Optional) IP address lease time in seconds. Minimum value is 60.
+        Maximum value is 4294967295. Default value is 86400
+        Note: This field is applicable only when type is server
+
+    display_name
+        Identifier to use when displaying entity in logs or GUI.
+
+    Example values:
+
+        .. code::
+
+            for dhcp-server-profiles:
```

Comment:
should these examples be updated for state files? The syntax is different from modules.

Reply:
during create dhcp, display_name will be same as dhcp_profile_id, so the create method of execution module does not require the display_name parameter.

during update dhcp, display_name can be updated, so the update method of execution module requires the display_name parameter.

In the state module, first we are checking if there is any existing dhcp profile for the given dhcp profile Id, if not exists we are creating new profile else we are updating the existing profile, thats why the syntax is different from modules

Reply:
will do this change

Reply:
updated in local branch. will push all changes together as new PR